### PR TITLE
📖 book: Update versions doc adding Kubernetes 1.28 + prowjob documentation

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Update book:
  
  * Update supported versions in `versions.md`
  * Update job documentation in `jobs.md`
  * Prior art: [🌱 Update versions doc adding Kubernetes 1.25 #7194](https://github.com/kubernetes-sigs/cluster-api/pull/7194) [📖 docs update jobs.md #7196](https://github.com/kubernetes-sigs/cluster-api/pull/7196)

TODO

- [ ] Wait for merge of https://github.com/kubernetes/test-infra/pull/30347

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #8708

### CODE REVIEW

File: docs/book/src/reference/jobs.md [hunk 1]
```diff
@@ -137,6 +137,7 @@ Prow Periodics:
 [periodic-cluster-api-e2e-workload-upgrade-1-24-1-25-main]: https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api#capi-e2e-main-1-24-1-25
```
Comment (sbueringer) [inline docs/book/src/reference/jobs.md:137]: We can delete [periodic-cluster-api-e2e-workload-upgrade-1-22-1-23-main] now

**Comment by sbueringer** [comment]
Btw I think /hold is enough you don't have to keep it in draft state. Makes it slightly clearer that the PR can already be reviewed + we can run tests

**Comment by chrischdi** [comment]
/hold

Waiting for
- https://github.com/kubernetes/test-infra/pull/30347

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

**Comment by sbueringer** [comment]
Looks good!

/approve

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Pull request merged**